packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,339 @@
+    mocks.getNetworkClientById ||
+    jest.fn().mockReturnValue({
+      configuration: {
+        chainId: '0x1',
+      },
+    });
+  messenger.registerActionHandler(
+    'NetworkController:getNetworkClientById',
+    mockGetNetworkClientById,
+  );
+
+  const mockGetProviderConfig =
+    mocks.getProviderConfig ||
+    jest.fn().mockReturnValue({
+      providerConfig: {
+        chainId: '0x1',
+      },
+    });
+  messenger.registerActionHandler(
+    'NetworkController:getState',
+    mockGetProviderConfig,
+  );
+
+  const mockEnqueueRequest = jest.fn().mockImplementation((cb) => cb());
+  messenger.registerActionHandler(
+    'QueuedRequestController:enqueueRequest',
+    mockEnqueueRequest,
+  );
+
+  const mockAddRequest = mocks.addRequest || jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    'ApprovalController:addRequest',
+    mockAddRequest,
+  );
+
+  const mockSetProviderType = jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    'NetworkController:setProviderType',
+    mockSetProviderType,
+  );
+
+  const mockSetActiveNetwork = jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    'NetworkController:setActiveNetwork',
+    mockSetActiveNetwork,
+  );
+
+  const mockSetNetworkClientIdForDomain = jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain,
+    mockSetNetworkClientIdForDomain,
+  );
+
+  return {
+    getProviderConfig: mockGetProviderConfig,
+    getNetworkClientById: mockGetNetworkClientById,
+    enqueueRequest: mockEnqueueRequest,
+    addRequest: mockAddRequest,
+    setActiveNetwork: mockSetActiveNetwork,
+    setProviderType: mockSetProviderType,
+    setNetworkClientIdForDomain: mockSetNetworkClientIdForDomain,
+  };
+};
+
+const requestDefaults = {
+  method: 'doesnt matter',
+  id: 'doesnt matter',
+  jsonrpc: '2.0' as const,
+};
+
+const noop = jest.fn();
+
+describe('createQueuedRequesMitddleware', () => {
+  it('should not enqueue the request when useRequestQueue is false', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware(messenger, () => false);
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      ...requestDefaults,
+      origin: 'example.com',
+    };
+
+    await new Promise((resolve) => middleware(req, {} as any, resolve, noop));
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  it('should not enqueue the request when there is no confirmation', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware(messenger, () => true);
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      ...requestDefaults,
+      origin: 'example.com',
+      method: 'eth_chainId',
+    };
+
+    await new Promise((resolve) => middleware(req, {} as any, resolve, noop));
```

Comment:
worth making sure the expected value is returned as well?

Reply:
The result here doesn't matter, we never return a result, nor is there any 'next' middleware in this test that will create a result. All we care about is that the request is not queued.